tox.ini

```
@@ -1,54 +1,52 @@
 [tox]
-envlist = py38-django{22,30,31,32}-celery{44,50}-drf{312,latest},quality,docs
+envlist = py38-django{32,40}-celery{44,50}-drf{312,latest},quality,docs
 
 [testenv]
-deps =
-	django22: Django>=2.2,<2.3
-	django30: Django>=3.0,<3.1
-	django31: Django>=3.1,<3.2
-	django32: Django>=3.2,<3.3
-	drf312: djangorestframework>=3.12,<3.13
-	drflatest: djangorestframework
-	celery44: -r{toxinidir}/requirements/celery44.txt
-	celery50: -r{toxinidir}/requirements/celery50.txt
-	-r{toxinidir}/requirements/test.txt
-commands =
-	python -Wd -m pytest {posargs}
+deps = 
+    django32: Django>=3.2,<4.0
+    django40: Django>=4.0,<4.1
+    drf312: djangorestframework>=3.12,<3.13
+    drflatest: djangorestframework
+    celery44: -r{toxinidir}/requirements/celery44.txt
+    celery50: -r{toxinidir}/requirements/celery50.txt
+    -r{toxinidir}/requirements/test.txt
+commands = 
+    python -Wd -m pytest {posargs}
 
 [testenv:docs]
-allowlist_externals =
-	make
-	rm
-deps =
-	-r{toxinidir}/requirements/doc.txt
-commands =
-	doc8 --ignore-path docs/_build --ignore-path docs/rest_api.rst README.rst docs
-	rm -f docs/user_tasks.rst
-	rm -f docs/modules.rst
-	make -C docs clean
-	make -C docs html
-	python setup.py sdist bdist_wheel
-	twine check dist/*
+allowlist_externals = 
+    make
+    rm
+deps = 
+    -r{toxinidir}/requirements/doc.txt
+commands = 
+    doc8 --ignore-path docs/_build --ignore-path docs/rest_api.rst README.rst docs
+    rm -f docs/user_tasks.rst
+    rm -f docs/modules.rst
+    make -C docs clean
+    make -C docs html
+    python setup.py sdist bdist_wheel
+    twine check dist/*
 
 [testenv:quality]
-setenv=
+setenv = 
     DJANGO_SETTINGS_MODULE=test_settings
-allowlist_externals =
-	make
-	rm
-	touch
-deps =
-	-r{toxinidir}/requirements/doc.txt
-	-r{toxinidir}/requirements/quality.txt
-	-r{toxinidir}/requirements/test.txt
-commands =
-	touch tests/__init__.py
-	pylint user_tasks
-	pylint tests
-	pylint schema
-	rm tests/__init__.py
-	pycodestyle schema tests user_tasks
-	pydocstyle schema tests user_tasks
-	isort --check-only --diff --recursive schema tests user_tasks manage.py setup.py test_settings.py
-	make help
+allowlist_externals = 
+    make
+    rm
+    touch
+deps = 
+    -r{toxinidir}/requirements/doc.txt
+    -r{toxinidir}/requirements/quality.txt
+    -r{toxinidir}/requirements/test.txt
+commands = 
+    touch tests/__init__.py
+    pylint user_tasks
+    pylint tests
+    pylint schema
+    rm tests/__init__.py
+    pycodestyle schema tests user_tasks
+    pydocstyle schema tests user_tasks
+    isort --check-only --diff --recursive schema tests user_tasks manage.py setup.py test_settings.py
```

`recursive` flag isn't needed anymore with latest `isort` versions now.
```suggestion
    isort --check-only --diff schema tests user_tasks manage.py setup.py test_settings.py
```